features/dd-sdk-android-trace/src/main/java/com/datadog/opentracing/PendingTrace.java

```diff
@@ -178,25 +180,27 @@ public void cancelContinuation(final ContinuableScope.Continuation continuation)
   }
 
   private void expireReference() {
-    final int count = pendingReferenceCount.decrementAndGet();
-    if (count == 0) {
-      write();
-    } else {
-      if (tracer.getPartialFlushMinSpans() > 0 && size() > tracer.getPartialFlushMinSpans()) {
-        synchronized (this) {
-          if (size() > tracer.getPartialFlushMinSpans()) {
-            final DDSpan rootSpan = getRootSpan();
-            final List<DDSpan> partialTrace = new ArrayList(size());
-            final Iterator<DDSpan> it = iterator();
-            while (it.hasNext()) {
-              final DDSpan span = it.next();
-              if (span != rootSpan) {
-                partialTrace.add(span);
-                completedSpanCount.decrementAndGet();
-                it.remove();
+    synchronized(this) {
```

Comment:
do we need this one? it seems that code between L184 and L186 should _probably_ be safe without it (it is using atomic primitives and references other classes only). And code below L189 already has `synchronized` block. Do we have a test failure without that change?

Reply:
Good call.